src/shogun/features/CombinedDotFeatures.cpp

```diff
@@ -19,9 +23,10 @@ CCombinedDotFeatures::CCombinedDotFeatures() : CDotFeatures()
 	update_dim_feature_space_and_num_vec();
 }
 
-CCombinedDotFeatures::CCombinedDotFeatures(const CCombinedDotFeatures & orig)
-: CDotFeatures(orig), num_vectors(orig.num_vectors),
-	num_dimensions(orig.num_dimensions)
+CCombinedDotFeatures::CCombinedDotFeatures(const CCombinedDotFeatures& orig)
+    : CDotFeatures(orig), feature_array(orig.feature_array),
+      feature_weights(orig.feature_weights), num_vectors(orig.num_vectors),
+      num_dimensions(orig.num_dimensions)
 {
 	init();
```

Comment:
please note that aything that is initialised in `init` over-writes the initialization fields used above